### PR TITLE
Syncing metadata links in redis

### PR DESCRIPTION
**Pull request by hamzatrq**
Syncing the metadata uris in redis

### CODE REVIEW

**Comment by hamzatrq** [comment]
This PR needs to be reviewed and merged for the latest PR for backend-api to work properly.

File: getTokenIds-test.js [hunk 1]
```diff
@@ -1,26 +1,23 @@
...
+    const uris = await getTokenURIs("rinkeby", tokenIDs);
+    uris.forEach(async uri => {
+        try {
+            await putRedisItem(`${uri.tokenId}`, `${uri.tokenURI}`, redisPrefixes[config.contractName]+'uris');
+        } catch (error) {
+            console.log(error);
+        }
     });
+    console.log(redisPrefixes[config.contractName]+'uris');
+    fetchedURI = Object.values((await getRedisItem('2', redisPrefixes[config.contractName]+'uris')).Item).join('');
```
Comment (avaer) [inline getTokenIds-test.js:19]: This is a big mess. I have no idea what this is supposed to be, or what these mysterious constants and coersions are.

Can this be separate lines?

File: tokens.js [hunk 2]
```diff
@@ -954,39 +955,68 @@ const getAllWithdrawsDeposits = contractName => async chainName => {
...
+  const contract = blockchain.contracts["rinkeby"].rinkebyWebaverseERC721;
+
+
+  tokenURIs = [];
+  for (let tokenId in tokenIDs) {
+    try {
+      const tokenURI = await contract.methods.tokenURI(tokenId).call();
+      // The URI can be anything, since public is allowed to mint tokens.
+      // We need to solve it but for now removing tokens with invalid IPFS uri
+      if (isIPFS.url(tokenURI)) {
```
Comment (avaer) [inline tokens.js:999]: Definitely do not resolve the token on the Redis side.

File: package.json [hunk 3]
```diff
@@ -21,6 +21,7 @@
   "dependencies": {
     "express": "^4.17.1",
     "forever": "^3.0.4",
+    "is-ipfs": "^6.0.2",
```
Comment (avaer) [inline package.json:24]: Remove this.

File: cache.js [hunk 4]
```diff
@@ -160,19 +160,31 @@ async function initAccountCache({chainName}) {
 }
 async function initTokenIdsCache() {
   let tokenIdOwners = await getTokenIDs(config.contractName)(config.network);
+  const tokenIDs = Object.keys(tokenIdOwners);
+  const uris = await getTokenURIs("rinkeby", tokenIDs);
+  uris.forEach(async uri => {
+    try {
+      await putRedisItem(`${uri.tokenId}`, `${uri.tokenURI}`, redisPrefixes[config.contractName] + 'uris');
```
Comment (avaer) [inline cache.js:167]: What is the point of this weird stringification? Are the types so unclear that there is a coersion needed?

**Review by avaer — CHANGES_REQUESTED** [review]
Major code and design problems.

IPFS should have nothing to do with the redis server, and the code is a mess in the typing.

**Pull request closed without merging**